expression/builtin_time_vec.go

```diff
@@ -541,11 +541,41 @@ func (b *builtinSubDateStringStringSig) vecEvalTime(input *chunk.Chunk, result *
 }
 
 func (b *builtinQuarterSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalInt evals QUARTER(date).
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_quarter
 func (b *builtinQuarterSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		return err
```

Comment:
Should we also check `handleInvalidTimeError` here?

Reply:
It's the `args[0]`'s responsibility.